### ISSUE TITLE
Connection refused on default namespace

### ISSUE THREAD

**Issue by Vihar1** [terraform-google-modules/terraform-google-kubernetes-engine#1653]
### TL;DR

Workload Identity service account is created but TF throws error.

### Expected behavior

TF runs without error.

### Observed behavior

google_container_cluster.key-test-gke-cluster[0]: Creation complete after 11m18s [id=projects/test-12/locations/europe-west3/clusters/key-test-gke-cluster]
╷
│ Error: Post "http://localhost/api/v1/namespaces/default/serviceaccounts": dial tcp [::1]:80: connect: connection refused
│
│   with module.workload-identity-user.kubernetes_service_account.main[0],
│   on .terraform/modules/workload-identity-user/modules/workload-identity/main.tf line 50, in resource "kubernetes_service_account" "main":
│   50: resource "kubernetes_service_account" "main" {
│
╵

Also, I am not sure this IAM Binding (roles/iam.workloadIdentityUser) between the service accounts is created.

### Terraform Configuration

```hcl
module "workload-identity-user" {
  source     = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
  name       = "wi-test-user"
  namespace  = "default"
  project_id = var.project_id
  roles      = ["roles/storage.admin", "roles/compute.admin"]
}
```

### Terraform Version

```sh
Terraform v1.4.6
on darwin_arm64
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/google v4.66.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.20.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
```
